packages/web/src/js/Dropdown.ts

```diff
@@ -0,0 +1,208 @@
+
+  constructor(element: SpiritElement) {
+    super(element);
+    this.target = SelectorEngine.findOne(`#${this.element.dataset.target}`);
+    this.reference = this.findReferenceElement();
+    this.state = {
+      open: false,
+    };
+    this.options = {
+      placement: placementKeys['bottom-start'],
+      offset: 0,
+      padding: 0,
+      autoClose: true,
+      autoWidth:
+        this.element.dataset.autowidth ||
+        this.element.dataset.autowidth === '' ||
+        this.element.dataset.autowidth === 'true',
+    };
+    this.autoUpdateOptions = {
+      ancestorScroll: true,
+      ancestorResize: true,
+      elementResize: true,
+      animationFrame: false,
+    };
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  getOptions() {
+    const options = { ...this.options };
+    const dataset = this.element?.dataset;
+    const optionsPlacement = dataset?.placement;
+    const optionsOffset = dataset?.offset;
+    const optionsPadding = dataset?.padding;
+    const optionsAutoClose = dataset?.autoclose;
+
+    if (optionsPlacement) options.placement = optionsPlacement;
+    if (optionsOffset) options.offset = Number(optionsOffset);
+    if (optionsPadding) options.padding = Number(optionsPadding);
+    if (optionsAutoClose) options.autoClose = Boolean(!optionsAutoClose);
+
+    return options;
+  }
+
+  findReferenceElement() {
+    let anchor = this.element;
+    const { reference } = this.element.dataset;
+
+    if (reference) {
+      if (
+        reference === 'parent' &&
+        this.element.parentElement &&
+        this.element.parentElement.classList.contains(CLASSNAME_DROPDOWN)
+      ) {
+        anchor = this.element.parentElement;
+      } else if (reference.match(/(#)/g)) {
+        anchor = SelectorEngine.findOne(reference);
+      }
+    }
+
+    return anchor;
+  }
+
+  updateTriggerElement(open: boolean = this.state.open) {
+    this.element.classList.toggle(CLASSNAME_EXPANDED, open);
+    this.element.setAttribute(ARIA_EXPANDED_ATTRIBUTE, open);
+    this.element.setAttribute(ARIA_CONTROLS_ATTRIBUTE, this.element.dataset.target);
+  }
+
+  updateTargetElement(open: boolean = this.state.open) {
+    this.target?.classList.toggle(CLASSNAME_OPEN, open);
+    if (this.options.autoWidth && this.target)
+      Object.assign(this.target.style, { width: `${this.element.offsetWidth}px` });
+  }
+
+  async alignTargetElement() {
+    const options = this.getOptions();
+    const { placement } = options;
+    const middleware = [
+      offset(options.offset),
+      flip(),
+      shift({
+        padding: options.padding,
+      }),
+    ];
+
+    if (this.reference && this.target) {
+      const assignPosition = (x: number, y: number) => {
+        this.target &&
+          Object.assign(this.target.style, {
+            left: `${x}px`,
+            top: `${y}px`,
+          });
+      };
+      const positions = computePosition(this.reference, this.target, { placement, middleware });
```

Comment:
```suggestion
      const {x, y} = await computePosition(this.reference, this.target, { placement, middleware });
```

https://github.com/floating-ui/floating-ui/blob/ae2e56e751d247d81b162112cd46238c5290e685/packages/core/test/computePosition.test.ts#L16